### ISSUE TITLE
setup guide for nextjs?

### ISSUE THREAD

**Issue by Elon-Mask12** [scribeocr/scribe.js#7]
hey, anyone managed to get it working in nextjs pages dir? 
heres what ive tried:
imported as:
```ts
// @ts-ignore
import scribe from "scribe.js-ocr/scribe.js";
```

```
//other code
await scribe.init({ ocr: true, font: true });
scribe.extractText(files).then((res: any) => console.log(res));
```
i see error saying: 
"Import trace for requested module:
./node_modules/.pnpm/scribe.js-ocr@0.2.6_typescript@5.3.3/node_modules/scribe.js-ocr/scribe.js
./src/components/upload-file-modal.tsx
./src/pages/f/index.tsx
 ○ Compiling /not-found ...
 ⨯ ./node_modules/.pnpm/scribe.js-ocr@0.2.6_typescript@5.3.3/node_modules/scribe.js-ocr/js/containers/fontContainer.js:14:26
  Module not found: Can't resolve 'module'"

i updated my next.config to pass custom webpack configs similar to the one in the example app, but it doesnt work.
also tried passing it in:
```
  experimental: {
    serverComponentsExternalPackages: [
      "scribe.js-ocr",
      "scribe.js-ocr/scribe.js",
    ],
  },
  ```
  
  this doesnt work either, any help is appreciated thanks.

**Comment by Balearica** [comment]
I generated a repo using `npx create-next-app@latest`, added some basic scribe.js code, and updated the webpack configuration, and everything seemed to run as expected.  The repo is [here](https://github.com/scribeocr/scribe.js-example-next.js).

**Comment by Elon-Mask12** [comment]
holy fuck, that works. also thanks a fuckton for building such a library, and even replying to my query. <3
i got it workin on my next app, but im having this issue where im writing this function which takes a pdf in File type and then return the ocr-ed version of the pdf in file type. So this will be the code right?


```typescript
await scribe.init({ ocr: true, font: true });
scribe.opt.displayMode = "invis";

await scribe.importFiles(
  files,
);
await scribe.recognize({
  mode: "quality",
  langs: ["eng"],
  modeAdv: "combined",
  vanillaMode: true,
  combineMode: "data",
});
const data = await scribe.exportData("pdf");

const blob = new Blob([data], { type: "application/pdf" });
const file = new File([blob], "test.pdf", {
  type: "application/pdf",
});
return [file];
````
or am i doing something repeatedly, the docs seems a bit hard to understand. 

regarding the options i passed in for each, its just something that worked for me when i was playing around with https://scribeocr.com . am i doing something wrong?
thanks :)

**Comment by zeus-12** [comment]
used the `next build` version, and scribeocr.com is significantly better. 
i uploaded a 34 page file, with plenty of images-took around `40s` in scribeocr, and around 3 minutes in my app. clearly doing something wrong, also any way i could access the progress? 

thanks.

**Comment by zeus-12** [comment]
used the `next build` version. tested the same pdf on both, and scribeocr.com is significantly better. 
i uploaded a 34 page file, with plenty of images, took around `40s` in scribeocr, and around 3 minutes in my app. clearly doing something wrong, also any way i could access the progress (the percentage of completion)? kinda debating now whether ocr on browser is a good idea.

thanks.

**Comment by Balearica** [comment]
> i uploaded a 34 page file, with plenty of images, took around 40s in scribeocr, and around 3 minutes in my app

I checked that the Next.js demo I linked has runtime similar to scribeocr.com, and it looks like it does, so I think any disparity is caused by something specific to your project.

> also any way i could access the progress (the percentage of completion)

The scribeocr.com website is simply one user of the scribe.js package--everything it does can be implemented by other users.  While not all of the features it uses are fully documented, you can check [that codebase](https://github.com/scribeocr/scribeocr) to see how specific features work.  In this case the relevant code for incrementing a progress bar after each page is [here](https://github.com/scribeocr/scribeocr/blob/8d68969821989eaae18a4919e6d9fd622498eb4e/main.js#L68-L104).

**Comment by zeus-12** [comment]
You're right, that improved a lot in prod-setting, and getting super-close results. Thanks for building this <3

**Comment by gcphost** [comment]
Hi there! I'm trying this with nextjs app dir, when setting the next config to set process as undefined I am no longer able to get environment variables, for example, from a middleware.

```TypeError: Cannot set properties of undefined (setting 'env')```

Is there any other solution?
